lib/commonAWS.js

```diff
@@ -1,16 +1,60 @@
 'use strict';
 const AWS = require('aws-sdk');
+const ini = require('ini-config-parser');
+const os = require('os');
 
-module.exports = (argv) => {
-  const key = argv.access_key ? argv.access_key : process.env.AWS_ACCESS_KEY_ID;
-  const secret = argv.secret_key ? argv.secret_key : process.env.AWS_SECRET_ACCESS_KEY;
-  const region = argv.region ? argv.region : process.env.AWS_DEFAULT_REGION;
-  const profile = argv.profile ? argv.profile : 'default';
-  if (key && secret && profile === 'default') {
-    console.log(`initializing AWS with ${key} ${secret} ${region}  ${profile}`);
-    AWS.config.update({ accessKeyId: key, secretAccessKey: secret, region });
+const fallbackAWSRegion = 'us-east-1';
+
+// the order of precedence for setting the credentials is:
+// 1. --access_key and --secret-key command-line parameters
+// 2. AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY environment variables
+// 3. the ~/.aws/credentials file
+const initCustomCredentials = (params) => {
+  const key = params.access_key ? params.access_key : process.env.AWS_ACCESS_KEY_ID;
+  const secret = params.secret_key ? params.secret_key : process.env.AWS_SECRET_ACCESS_KEY;
+  const profile = params.profile ? params.profile : 'default';
+  // if they provided an access key and secret key, load this:
+  if (key && secret) {
+    AWS.config.update({ accessKeyId: key, secretAccessKey: secret });
   } else {
     AWS.config.credentials = new AWS.SharedIniFileCredentials({ profile });
   }
-  return new AWS.CloudWatchLogs({ region });
+};
+
+const extractRegionFromConfigFile = (params, callback) => {
+  const sharedAWSConfigFilePath = os.homedir() + `/.aws/config`;
+  const data = ini.parse(sharedAWSConfigFilePath);
+  const profile = params.profile ? params.profile : 'default';
+  // some config files use this format:
+  if (data[`profile ${profile}`]) {
+    return callback( data[profile].region);
+  } else {
+    return callback( data[profile].region);
+  }
+};
+
+// the order of precedence for setting the AWS region is:
+// 1. --region command line parameter
+// 2. AWS_DEFAULT_REGION environment variable
+// 3. the ~/.aws/config file
+// 4. 'us-east-1'
+const initRegion = (params, callback) => {
```

Comment:
it doesn't look like anything is async here. Do you need a callback or can you just return normally?